### DISCUSSION TITLE
Early Access to v3.0.0.0 available for testing!

### DISCUSSION THREAD

**Discussion by Ross-Patterson** [Ross-Patterson/Portals-Desktop-Organization#107] [Announcements]
If you want to help me with testing effort, early access to v3.0.0.0 is available:
{Link Redacted}

**EDIT:**
Based on @DarkAoRaidenX feedback below (thanks!) - I have added to the installer to check for .NET 5 Desktop Runtime and download/install it if required.
{Link Redacted}

**EDIT 2:**
Based on all of the feedback here, I have added:
- Icons support for .URL files i.e. steam.
- Added Open Settings option to context menu of Taskbar Icon
- Made settings open if taskbar icon is double clicked.
 
New Link (v3.0.0.2):
{Link Redacted}

**EDIT3:**
Fixed custom sorting not saving:
New Link (v3.0.0.3):
{Link Redacted}

**EDIT4:**
- Downgraded to .NET 4.8
- Stripped out bloaty Material UI Library
- Fixed Launch on Startup
- Added Customize button on Main Settings window table
New Link (v3.0.0.4):
{Link Redacted}

Please provide the feedback on this thread.

Thanks,
Ross

**Comment by DarkAoRaidenX** [comment]
Yo, so i tried to install and i get this error. I must install .NET and i click the Yes button to download it doesn't work nothing happened? it didn't take to some where to download it or didn't see any progress happening and i let it sit for a few mins.

so i manually installed it but then the installer says i already have the latest installed?

[Got the download from here](https://dotnet.microsoft.com/en-us/download/dotnet-framework)

![03 04 2023 12 55 15](https://user-images.githubusercontent.com/101914106/222924136-9dff4dbc-ae50-46d0-8d46-e6499be7ef3c.png)
![03 04 2023 13 02 27](https://user-images.githubusercontent.com/101914106/222924138-b5bc0962-7454-44af-8e22-1e94005100d1.png)

**Comment by Ross-Patterson** [comment]
I am surprised that clicking yes does nothing - this is actually a windows popup and nothing I added to the app. 

Anyway, you can get the download from here:
https://dotnet.microsoft.com/en-us/download/dotnet/5.0

![image](https://user-images.githubusercontent.com/66630787/222924439-77012de4-a2c4-4094-86b3-47e2b53f6f6d.png)

**Comment by DarkAoRaidenX** [comment]
I tried 7.0.2/.3 as on 5 it says its end of life and im still getting that error. I even fully uninstalled. Do i need 5 specifically?

"This release has reached end of life, meaning it is no longer supported. We recommend moving to a supported release, such as [.NET 7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0). See our [support policy](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core) for more details."

**Comment by Ross-Patterson** [comment]
Yes, at this time .NET 5 Desktop Runtime is required.

**Comment by DarkAoRaidenX** [comment]
Ok i found the issues...
1st - did the SDK
2nd - did the RunTime
3rd - did the Desktop RunTime and it worked i'm on 7.03 right now and it is working.

 I can now say i'm glad you're back and updated the app i was getting worried haha
 EDIT: I didnt mean like... it sound to mean just meant wanted to say it first but had to deal with the issues first Dx
 
![03 04 2023 18 10 51](https://user-images.githubusercontent.com/101914106/222934651-b0763dab-c266-4bf1-a77b-01520ff497c1.png)
![03 04 2023 18 15 07](https://user-images.githubusercontent.com/101914106/222934755-c993af71-ef61-4e43-ac73-43851c9b38fc.png)

**Comment by DarkAoRaidenX** [comment]
[v3.0.0.1](https://downloads.portals-app.com/installers%2F3.0.0.1%2Fportals_installer_v3-0-0-1.exe?alt=media) still working with 7.03 .NET 

EDIT: I did a PC restart test and yeah seems like my issue is fixed values stayed. :D
![image](https://user-images.githubusercontent.com/101914106/222940738-d542cd38-245f-4b2e-81a4-b4fa459ed4d4.png)

**Reply by Ross-Patterson** [reply]
This is good news!

**Comment by darxian** [comment]
When dragging to a corner, seems to want to use the feature in windows to resize itself. Cannot put it to a corner anymore. Also some icons not working for some reason. Also the indocator of window size being made stays there even when accidentally dragging locked portal in the corner or side with the snap feature enabled.
![image](https://user-images.githubusercontent.com/104892211/222959430-79ed83d4-b29f-49c1-a531-b086e70ef320.png)

Also the previous version had a better settings imo.

**Reply by Ross-Patterson** [reply]
As long as the mouse pointer isn't near the corner it should work.

Here is a window right in the top corner of my screen:
![image](https://user-images.githubusercontent.com/66630787/222965733-1150f5cc-3bf8-40e6-b263-d5c9ebbcb358.png)

The snapping is determined by the proximity of the mouse pointer to the corner, not the window itself. So drag the window from a different position on the header.

What did you like about the previous settings view?

**Reply by darxian** [reply]
It was easier to access from the tray and seemed to be more customizable rather than individually clicking each portal for it's setting. I mean the new setting clicking a portal is nice to only have it's own setting there, but it should have a general settings option in tray, that allows you to go to the specific portal setting like it does in this version.

**Reply by Ross-Patterson** [reply]
You can still access it by just single clicking the tray icon. I will add in the option again - as it seems to be causing confusion

**Reply by darxian** [reply]
Ah, i just realised that, i thought i clicked it but it didn't open, thought the settings disappeared. It is fine.

**Comment by reffu42** [comment]
.url files have blank icons. If you look in the folder, the icons are all visible, but in the portal they're all blank. This is particularly an issue for steam games, which use a .url file pointing to `steam://` for most desktop shortcuts.
![image](https://user-images.githubusercontent.com/1823982/224225032-c968eb16-a05a-44a9-bced-272ffd22394b.png)
![image](https://user-images.githubusercontent.com/1823982/224225092-6696fd9b-bf9a-429b-adf7-716e19b08b5f.png)

**Reply by Ross-Patterson** [reply]
Knowing that the issue is related steam:// is very helpful, will investigate

**Reply by nathanbain** [reply]
![Cropped](https://user-images.githubusercontent.com/210289/226150052-9b7b1d6d-3431-47f7-bdf4-eb78e9a13dfb.png)

Same problem for me with Steam and Epic store game shortcuts. Xbox games are OK.

**Reply by Ross-Patterson** [reply]
Turns out icons for .URL files are handled differently, I have got it working locally in my machine now:
![image](https://user-images.githubusercontent.com/66630787/226170963-55d5b4f5-4ca5-49de-8cb0-1e0258ebf77c.png)

**Reply by Ross-Patterson** [reply]
@reffu42 @nathanbain 
v3.0.0.2 link now available above - please check if the Steam icons are working

**Reply by reffu42** [reply]
@Ross-Patterson Yup, icons are working now. Thanks for fixing that right away.

**Reply by nathanbain** [reply]
@Ross-Patterson Same here - all working perfectly now. Thankyou!

**Comment by reffu42** [comment]
There's no easy way to access settings. Not sure if this just isn't implemented yet, but right-clicking on the system tray icon only gives the option to exit, and double-clicking does nothing. In order to access settings I have to triple-click the tray-icon in order to get settings to open.
![image](https://user-images.githubusercontent.com/1823982/224225666-d440c23a-6c06-4dc5-85ee-8f7535008db1.png)

**Reply by pr3dd1** [reply]
I have tested 3.0.0.1 and must say that I didn't see the option to hide icon texts. Please let me know where to look.
Additional minor bugs:
The 2.x config was not loaded by 3.x, so I had to reconfigure. 
In addition I have noticed that the transparency option was gone.
Reverting back to 2.x invalidated the registration.

Best regards

**Reply by reffu42** [reply]
I assume you meant to reply to the discussion, and not my comment, but I can at least answer the transparency question. If you go to the Color select and click the dropdown button, one of the options is "A" which means "Alpha" or transparency. Lowering that changes how opaque vs transparent the background color is:
![image](https://user-images.githubusercontent.com/1823982/224348332-f774f923-32aa-47ab-9181-06ef95dac03a.png)

**Reply by reffu42** [reply]
Also, for the icon text, if you go to "Styling" and "Portal Content", there should be an option called "Item Text Height". Setting that to 0 hides the icon text

**Reply by Ross-Patterson** [reply]
@reffu42 You can still access the settings by just single clicking the tray icon. I will add in the option again - as it seems to be causing confusion

**Reply by Ross-Patterson** [reply]
@pr3dd1 transparency is now part of the colour picker, rather than being separate.

**Reply by reffu42** [reply]
> @reffu42 You can still access the settings by just single clicking the tray icon. I will add in the option again - as it seems to be causing confusion

That would do it, single clicking works. Thanks for the clarification. 

 I think for me, single-clicking a tray icon is not intuitive, since most other programs require a double-click in order to open their main window/settings. It's weird that single-clicking and triple-clicking work, but double-clicking doesn't. I suppose triple-click is just a double-click (do nothing), followed by a single-click (open settings). I'd suggest enabling double-click to open as an option as well,

**Reply by Ross-Patterson** [reply]
@reffu42 I have added double click support in 3.0.0.2 (link now above)

**Reply by reffu42** [reply]
@Ross-Patterson Thanks, Double-click is working for me now, and I see that "Open settings" was added to the right-click menu as well.

**Comment by pr3dd1** [comment]
Ok, thx for clarification - I will test again

10 Mar 2023 16:03:55 reffu42 ***@***.***>:

**Comment by theBoatman** [comment]
In v3.0.0.1 there seems to be a problem with the custom sort order. On every restart the icons are sorted by name again and not the custom order I configured before.

**Reply by Ross-Patterson** [reply]
I will check on that

**Reply by Ross-Patterson** [reply]
try v3.0.0.3, link added above - custom sorting should save and reload now.

**Reply by theBoatman** [reply]
Works perfectly in v3.0.0.3, thanks.

**Comment by just4747** [comment]
I just installed 3.0.0.3 over 3.0.0.1 and it didn't update at all despite successful installation. Am I missing something?

EDIT: Nvm, had to manually close the program before installing. Assumed the installer took care of that on its own but guess not.

**Comment by darxian** [comment]
I can't seem to make Portals autostart when windows starts. Tried with task scheduler, enabling it in startup apps, putting in startup folder. Nothing seems to work.

**Reply by Max1234-Ita** [reply]
I tried too. the Launch on startup option is activated but seems to be disregarded.
Using Windows 10 Professional.

**Reply by Ross-Patterson** [reply]
Thanks for reporting - I have fixed this in 3.0.0.4 which I will post a link to soon.

Turns out upgrading to .net core 5 was not a good idea - it came with a lot of unforeseen annoyances. I have downgraded back to .net framework 4.8.1.

**Reply by Ross-Patterson** [reply]
@darxian @Max1234-Ita 
v3.0.0.4 Link Now Available above

**Reply by Max1234-Ita** [reply]
Thank you so much. I'll try it as soon as I have some time (lots of relatives coming to visit on Easter day 🙂)

**Comment by GlitchGamingX** [comment]
Been using the 3.0 Beta for a while now and I can say it works wonderfully. Animations are smooth and responsive. Icon Spacing has also been fixed. There is enough horizontal spacing between icons on the Default Medium size.

**Comment by darxian** [comment]
Can we also have a button in the main settings to access each portals individual setting? I happened to make one portal completely transparent with no title or anything and now i can't customize it. So i deleted and made new one.

**Reply by Ross-Patterson** [reply]
Yes I can accommodate that

**Reply by Ross-Patterson** [reply]
v3.0.0.4 Link Now Available above

**Reply by darxian** [reply]
Seems perfect. Haven't tested all yet, but if startup works and also the customize button works, will be more than happy. So far the best desktop organizer i've tested (tried all the popular ones, don't even come close). Also happy to see you putting effort in maintaining this free software, that should honestly be paid. But i will donate when i have the means, since you deserve!

**Reply by Ross-Patterson** [reply]
Thanks! I noticed one small issue with v3.0.0.4 - I have re-uploaded the installer so I would recommend downloading and installing again just to avoid any issues.

**Comment by wasted0101** [comment]
Semi related to the Steam issue reported. All my icons for (internet shortcuts) are showing blank in the portal. I am on 3.0.0.4 and i believe it was working before.
![Capture](https://user-images.githubusercontent.com/67345128/235955063-aab4ceb5-f7c2-4661-aaca-f4f8c2e15425.PNG)
![Capture1](https://user-images.githubusercontent.com/67345128/235955095-05101aa0-4113-4c03-a03d-61a195bc5bd6.PNG)

**Reply by wasted0101** [reply]
Even if can just show the browser icon, that would work

**Reply by Ross-Patterson** [reply]
Making it show the default browser icon was fairly easy:
Before:
![image](https://user-images.githubusercontent.com/66630787/236043692-6dcede0c-010b-43a9-ae56-e0e89114966a.png)


After:
![image](https://user-images.githubusercontent.com/66630787/236043579-98e2fc24-be56-4a72-8856-aa70cf9b3b62.png)

Will deploy in the next update.

**Reply by wasted0101** [reply]
thank you for such a quick response

**Reply by wasted0101** [reply]
Was this fix included in 3.0.1.0? If so this did not resolve my issue, if not i will wait for the next release.

**Reply by Ross-Patterson** [reply]
No it wasn't, it will be included in the next one.

**Comment by DarkAoRaidenX** [comment]
Very minor thing i noticed hovering over the Clock icon at the bottom in the window for Update Available, there is no hover text pop up? like the Download icon saying "Start Update". 

Best i can guess it's for update in X time frame or reminder or something?

![05 17 2023 17 00 02](https://github.com/Ross-Patterson/Portals-Desktop-Organization/assets/101914106/10a7d605-dd65-4094-b3c5-86cc5da45aa4)
